### PR TITLE
[WIP] Remove deprecated options, aliases and defaults

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Remove deprecations for 4.0.0.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
various modules and module utils

### CODE REVIEW

**Comment by felixfontein** [comment]
ready_for_review

File: plugins/modules/cloud/misc/proxmox.py [hunk 1]
```diff
@@ -168,16 +154,15 @@
...
-        user expects different behavior from proxmox by default or fill options which cause
-        problems when they have been set.
-      - The default value is C(compatibility), which will ensure that the default values
-        are used when the values are not explicitly specified by the user.
-      - From community.general 4.0.0 on, the default value will switch to C(no_defaults). To avoid
-        deprecation warnings, please set I(proxmox_default_behavior) to an explicit
-        value.
+      - Since community.general 4.0.0, various options have no default values anymore.
+        These caused problems when user expects different behavior from proxmox by default or fill options
+        which caused problems when they have been set.
```
Comment (Ajpantuso) [inline plugins/modules/cloud/misc/proxmox.py:159]:
```suggestion
      - As of community.general 4.0.0, various options no longer have default values.
        These default values caused problems when users expected different behavior from Proxmox 
        by default or filled options which caused problems when set.
```
A little more concise wording here makes sense to me.

File: plugins/modules/cloud/misc/proxmox_kvm.py [hunk 2]
```diff
@@ -489,18 +454,17 @@
...
-        user expects different behavior from proxmox by default or fill options which cause
-        problems when they have been set.
-      - The default value is C(compatibility), which will ensure that the default values
-        are used when the values are not explicitly specified by the user.
-      - From community.general 4.0.0 on, the default value will switch to C(no_defaults). To avoid
-        deprecation warnings, please set I(proxmox_default_behavior) to an explicit
-        value.
+      - Since community.general 4.0.0, various options have no default values anymore.
+        These caused problems when user expects different behavior from proxmox by default or fill options
+        which caused problems when they have been set.
```
Comment (Ajpantuso) [inline plugins/modules/cloud/misc/proxmox_kvm.py:459]:
```suggestion
     - As of community.general 4.0.0, various options no longer have default values.
        These default values caused problems when users expected different behavior from Proxmox 
        by default or filled options which caused problems when set.
```

Same here.

**Review by Ajpantuso — CHANGES_REQUESTED** [review]
Just some docs suggestions for the Proxmox modules, but otherwise looks good.

File: plugins/modules/cloud/misc/proxmox.py (hunk 1 above)
Comment (felixfontein) [reply plugins/modules/cloud/misc/proxmox.py]: Thanks, I added both.

File: plugins/modules/system/xfconf.py [hunk 3]
```diff
@@ -60,11 +60,10 @@
     version_added: 1.0.0
   disable_facts:
     description:
-    - For backward compatibility, output results are also returned as C(ansible_facts), but this behaviour is deprecated
-      and will be removed in community.general 4.0.0.
-    - This flag disables the output as facts and also disables the deprecation warning.
+    - The value C(false) is no longer allowed since community.general 4.0.0.
+    - This option will deprecated in a future version, and eventually be removed.
```
[the remark this answers is not in the document]
Comment (felixfontein) [reply plugins/modules/system/xfconf.py]: Ooops. Thanks!
